### ISSUE TITLE
Mouse Deceleration

### ISSUE THREAD

**Issue by DevastatioN1337** [chocolate-doom/chocolate-doom#595]
Using Logitech G9, faster you move the mouse the slower the doom guy turns.  Few others are having this issue as well.

grabmouse                     1
novert                        1
mouse_acceleration            1.000000
mouse_threshold               0

**Comment by WChrisK** [comment]
I have the same issues. The deceleration makes it near impossible to play properly online with anyone else. I use a roccat savu, and the above settings too.

**Comment by fragglet** [comment]
Do you have the same problem with other source ports, especially LibSDL-based ones like PrBoom+?

**Comment by DevastatioN1337** [comment]
Downloaded prboom+, definitely the same problem.  Any known way to resolve?

**Comment by haleyjd** [comment]
Given this is a high-end 1kHz polling rate mouse, my uneducated guess would be that the event queue gets wrapped by the input and drops events. Though, is this the queue in Doom, or the one in SDL itself that is the problem?

**Comment by WChrisK** [comment]
If it helps, I play with it at 250 Hz and 800 dpi, it can go up to 1000Hz if I want it to though. Same issue still happens... sadly I don't have a bad mouse to test this with.

**Comment by WChrisK** [comment]
Is there anything I can do to help test this? Or find a cause?

**Comment by fabiangreffrath** [comment]
IIRC, in each game tic, i.e. 35 times per second, chocolate-doom centers the virtual/hidden mouse cursor and measures how far it gets moved out of its centered position. It then calculates in-game movement from the respective distances in x- and y-direction and resets the cursor into its center position again.

If your drivers polls mouse movement several hundred times per second with several hundred dpi, I can imagine that your driver reports several thousand pixels movement and this high number overflows a buffer or gets misinterpreted in some other way. Sounds somehow naive but may be possible.

tl;dr Try reducing your mouse driver's frequency and dpi even further. ;)

**Comment by fragglet** [comment]
A thought: Is it possible that if the mouse movement is turned up really high, the invisible cursor gets to the edge of the screen within a single tic?

**Comment by fabiangreffrath** [comment]
If the mouse is polled 250 times per second, i.e. more than 7 times per game tic, with 800 dpi, my guess is "yes". ;)

**Comment by fragglet** [comment]
Worth noting, but the invisible cursor thing is actually kind of a hack - SDL allows sampling of raw mouse data but there was (still is?) a bug in SDL that causes problems with sampling it - I think there's a small lag in input that really good players can feel. The invisible cursor solution is a workaround from PrBoom+ that was important into Chocolate Doom.

**Comment by fx02** [comment]
That sounds pretty bad, now I understand some issues I receive for competition doom. Will SDL2 handle this better?

**Comment by 07416** [comment]
Same issue. Does `mouse_acceleration            1.000000` mean no acceleration? What `mouse_threshold` setting should I use when disabling acceleration (tried both 0 and 10)?

**Comment by AXDOOMER** [comment]
> Same issue. Does mouse_acceleration 1.000000 mean no acceleration? What mouse_threshold setting should I use when disabling acceleration (tried both 0 and 10)?

Check this out: https://www.chocolate-doom.org/wiki/index.php/Setup/Mouse_configuration#Mouse_motion

**Comment by 07416** [comment]
GZDoom/ZDoom don't have this issue, but I understand the aim of these source ports is completely different.

**Comment by AXDOOMER** [comment]
> GZDoom/ZDoom don't have this issue

I don't think there could be an issue regarding this with these ports. I think it's specific to Chocolate-Doom to emulate a DOS mouse driver. In ZDoom, you can only set the mouse sensitivity, there is no mouse acceleration, so there is no threshold too.

**Comment by CapnClever** [comment]
@07416 My recommendation to you is to try out [this test build of Chocolate Doom using SDL2](http://jmtd.net/tmp/chocolate-doom-sdl2-branch-win32-dbg.zip). The mouse code captures raw input much better from my experience: see if it resolves the issue. (Not that it means it'd be universally solved but it'd be another reason to expedite sdl2-branch.)

Not sure if this would be related, but do you have different experiences between fullscreen and windowed modes?

**Comment by WChrisK** [comment]
@CapnClever 

Confirming the SDL2 debug version is pleasant to use. I think the test build provided above solved it.

I use always use fullscreen.
